crates/eth-rpc/src/rpc.rs

```diff
@@ -0,0 +1,42 @@
+use jsonrpsee::RpcModule;
+use kakarot_rpc_core::client::config::{JsonRpcClientBuilder, StarknetConfig};
+use kakarot_rpc_core::client::KakarotClient;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+
+use crate::api::eth_api::EthApiServer;
+use crate::api::kakarot_custom_api::KakarotCustomApiServer;
+use crate::servers::eth_rpc;
+use crate::servers::eth_rpc::KakarotEthRpc;
+use crate::servers::kakarot_custom_rpc::{self, KakarotCustomRpc};
+
+pub struct KakarotRpc {
+    eth_rpc_module: RpcModule<KakarotEthRpc<JsonRpcClient<HttpTransport>>>,
+    kakarot_custom_rpc_module: RpcModule<KakarotCustomRpc<JsonRpcClient<HttpTransport>>>,
+}
+
+impl KakarotRpc {
+    pub fn new(starknet_config: StarknetConfig) -> Self {
+        let eth_rpc_module =
+            eth_rpc::KakarotEthRpc::new(Box::new(KakarotRpc::get_kakarot_client(starknet_config.clone()))).into_rpc();
+        let kakarot_custom_rpc_module =
+            kakarot_custom_rpc::KakarotCustomRpc::new(Box::new(KakarotRpc::get_kakarot_client(starknet_config)))
+                .into_rpc();
+
+        Self { eth_rpc_module, kakarot_custom_rpc_module }
+    }
+
+    pub fn get_rpc_module(&self) -> RpcModule<()> {
+        let mut rpc_module = RpcModule::new(());
+
+        rpc_module.merge(self.eth_rpc_module.clone()).unwrap();
```

Comment:
why do we unwrap here? is there a way around it?

Reply:
I am returning a Result now { I was hacking fast on the PR, and then pulled it out from draft on daily without remembering about it!  }

<img width="798" alt="image" src="https://github.com/kkrt-labs/kakarot-rpc/assets/41180869/6b3c2a32-aa64-46f5-8f4c-2d03ba4a5152">